### PR TITLE
e2e: extend ConsulTemplate test and fix flakiness

### PR DESCRIPTION
**Pull request by tgross**
Fixes https://github.com/hashicorp/nomad/issues/8702

Add service discovery integration to the existing consul-template E2E test,
and verify both service and key updates force re-rendering. Fixes flakiness by
using the longer default wait config we use elsewhere.

---

This test ensures we've hit a specific customer's reported use case issues, and also removes the last remaining gomega consumer in the E2E suite.

### CODE REVIEW

**Review by cgbaker — APPROVED** [review]
nice 👍

File: e2e/consultemplate/consultemplate.go [hunk 1]
```diff
@@ -30,123 +33,231 @@ func init() {
...
+	count := 3
+	job.TaskGroups[2].Count = &count
+	_, _, err = tc.Nomad().Jobs().Register(job, nil)
+	f.NoError(err, "could not register job")
 
-	jobs := nomadClient.Jobs()
-	// Stop all jobs in test
-	for _, id := range tc.jobIds {
-		jobs.Deregister(id, true, nil)
+	// wait for re-rendering
```
Comment (cgbaker) [inline e2e/consultemplate/consultemplate.go:199]: do we want a test that `noop` is honored?

Comment (tgross) [reply e2e/consultemplate/consultemplate.go:199]: Good call.

Comment (tgross) [reply e2e/consultemplate/consultemplate.go:199]: Fixed in 4fa768d

**Pull request merged**